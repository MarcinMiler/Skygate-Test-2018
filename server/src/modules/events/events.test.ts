import { Connection } from 'typeorm'
import { createTestConn } from '../../testUtils/createTestConn'
import { request } from 'graphql-request'

let conn: Connection
beforeAll(async () => {
    conn = await createTestConn(true)
})

afterAll(async () => {
    conn.close()
})

const url = process.env.TEST_HOST as string

const addEventMutation = `
    mutation {
        createEvent(
            title: "Color Festival"
            description: "Some event"
            organizer: "Somelab"
            location: "Warsaw"
            startDate: "11.07.2018 10:30"
            endDate: "11.07.2018 20:00"
            photo: "photo.jpeg"
            category: "Music") {
                id
                title,
                description,
                organizer,
                location,
                startDate,
                endDate,
                photo,
                category,
            }
    }
`
const eventsQuery = `
    {
        events {
            id
            title,
            description,
            organizer,
            location,
            startDate,
            endDate,
            photo,
            category,
        }
    }
`
it('should return events', async () => {
    await request(url, addEventMutation)
    await request(url, addEventMutation)

    const response = await request(url, eventsQuery)
    expect(response).toMatchSnapshot()
})
