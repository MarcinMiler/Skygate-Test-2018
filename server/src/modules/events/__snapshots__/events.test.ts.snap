// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return events 1`] = `
Object {
  "events": Array [
    Object {
      "category": "Music",
      "description": "Some event",
      "endDate": "11.07.2018 20:00",
      "id": 1,
      "lat": 50.5,
      "lng": 35.5,
      "location": "Warsaw",
      "organizer": "Somelab",
      "photo": "photo.jpeg",
      "startDate": "11.07.2018 10:30",
      "title": "Color Festival",
    },
    Object {
      "category": "Music",
      "description": "Some event",
      "endDate": "11.07.2018 20:00",
      "id": 2,
      "lat": 50.5,
      "lng": 35.5,
      "location": "Warsaw",
      "organizer": "Somelab",
      "photo": "photo.jpeg",
      "startDate": "11.07.2018 10:30",
      "title": "Color Festival",
    },
  ],
}
`;
